PWGHF/HFC/Tasks/taskCorrelationHfeHadrons.cxx

```diff
@@ -0,0 +1,304 @@
+  HistogramConfigSpec hDphiDetaClusterTrackSpec{HistType::kTH3F, {{400, -0.2, 0.2}, {400, -0.2, 0.2}, {280, 0, 70}}};
+  HistogramConfigSpec hPIDSpec{HistType::kTHnSparseD, {{500, 0.0, 50.0}, {500, 0., 50.}, {300, -15, 15}, {300, 0.0, 30.0}, {400, 0, 2}, {400, 0, 2}}};
+  HistogramConfigSpec hTrackAllInfoSpec{HistType::kTHnSparseD, {{500, 0, 160}, {300, -15, 15}, {500, 0., 50.}, {500, 0., 50.}, {100, -1.5, 1.5}, {100, 0, 7}, {2000, -1, 1}, {2000, -1, 1}, {3, 0, 3}}};
+  HistogramConfigSpec hTrackInfoSpec{HistType::kTHnSparseD, {{500, 0, 160}, {300, -15, 15}, {500, 0., 50.}, {500, 0., 50.}, {100, -1.5, 1.5}, {100, 0, 7}}};
+
+  HistogramRegistry registry{
+    "registry",
+    {{"hNevents", "hNevents", {HistType::kTH1F, {{3, 1, 4}}}},
+     {"hZvertex", "z vertex", {HistType::kTH1F, {{100, -20, 20}}}},
+
+     {"hHHCorrel", "Sparse for Dphi and Deta with Inclusive electron;p_{T}^{e} (GeV#it{/c});p_{T}^{h} (GeV#it{/c});#Delta#varphi;#Delta#eta;", hCorrelSpec},
+     {"hInclusiveEHCorrel", "Sparse for Dphi and Deta with Inclusive electron;p_{T}^{e} (GeV#it{/c});p_{T}^{h} (GeV#it{/c});#Delta#varphi;#Delta#eta;", hCorrelSpec},
+     {"hDEdxnSigmaPt", "Sparse TPC info; dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi; DcaXY;Dcaz;passEMcal;", hTrackAllInfoSpec},
+     {"hHadronInformation", "Sparse hadron info; dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi;", hTrackInfoSpec},
+     {"hClusterInformationBefore", "Cluster Info before match; Energy (GeV);#eta;#varphi", hClusterInfoSpec},
+     {"hClusterInformationAfter", "Cluster Info after match; Energy (GeV);#eta;#varphi", hClusterInfoSpec},
+     {"hPIDafterMatch", "PID Info after match;dE/dx;n#sigma;#it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});#eta;#varphi;", hTrackInfoSpec},
+     {"hPIDafterPIDcuts", "PID Info after PID cuts; #it{p} (GeV#it{/c});#it{p}_{T} (GeV#it{/c});n_{#sigma}^{e};GeV;M02;M20", hPIDSpec},
+     {"hClsTrkEtaPhiDiffTime", "ClsTrkEtaPhiDiffTime;#Delta#eta;#Delta#varphi;Sec;", hDphiDetaClusterTrackSpec}}};
+
+  void init(o2::framework::InitContext&)
+  {
+    registry.get<THnSparse>(HIST("hInclusiveEHCorrel"))->Sumw2();
+    registry.get<THnSparse>(HIST("hHHCorrel"))->Sumw2();
+    registry.get<THnSparse>(HIST("hDEdxnSigmaPt"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClusterInformationBefore"))->Sumw2();
+    registry.get<THnSparse>(HIST("hClusterInformationAfter"))->Sumw2();
+    registry.get<THnSparse>(HIST("hPIDafterMatch"))->Sumw2();
+    registry.get<THnSparse>(HIST("hPIDafterPIDcuts"))->Sumw2();
+    registry.get<THnSparse>(HIST("hHadronInformation"))->Sumw2();
+  }
+
+  // correlation function for electron hadron
+
+  void correlation(GTrk const& eTrack, GTrks const& assotrks, int sparseNo = -1)
+  {
+    if (sparseNo < 0 || sparseNo > 1) {
+      LOGF(info, "Error: pass sparse value either '0' or '1'");
+      return;
```

Comment:
Use a proper error message.
Does it make sense to just skip the function and continue running if this happens?

Reply:
Dear Vit
Thank you for your suggestions. This function calculates the correlation distribution and fills the output in the desired Sparse, therefore, it will make no sense to skip the function and continue running the task